kong/cmd/utils/inject_confs.lua

```diff
@@ -0,0 +1,97 @@
+local conf_loader = require "kong.conf_loader"
+local pl_path = require "pl.path"
+local pl_stringx = require "pl.stringx"
+local prefix_handler = require "kong.cmd.utils.prefix_handler"
+local log = require "kong.cmd.utils.log"
+local fmt = string.format
+
+local compile_nginx_main_inject_conf = prefix_handler.compile_nginx_main_inject_conf
+local compile_nginx_http_inject_conf = prefix_handler.compile_nginx_http_inject_conf
+local compile_nginx_stream_inject_conf = prefix_handler.compile_nginx_stream_inject_conf
+local prepare_prefix = prefix_handler.prepare_prefix
+
+local function load_conf(args)
+  -- retrieve default prefix or use given one
+  log.disable()
+  local conf = assert(conf_loader(args.conf, {
+    prefix = args.prefix
+  }))
+  log.enable()
+
+  if pl_path.exists(conf.kong_env) then
+    -- load <PREFIX>/kong.conf containing running node's config
+    conf = assert(conf_loader(conf.kong_env))
```

Comment:
Please note that in conf_loader, the vault PDK will be used to dereference the values inside the configuration fields. Imagine that you have a vault backend with a self-signed certificate and you have reference value in your kong configuration, it will probably not work before and after this patch, because in conf_loader's runtime, the self-signed certificate will not be used even if you specify it in the `lua_ssl_trusted_certificate`. 

We should also consider solving this problem in the PR. Just proposing an idea that comes to mind: maintaining a list of configuration fields that cannot be vault referenced, such as those "injected" fields in this PR: `lmdb_environment_path`, `lmdb_map_size`, `lua_ssl_verify_depth`, `lua_ssl_trusted_certificate` and `lua_ssl_protocols`. If we have such fields added in the future, we should also bring them into the list because those fields are going to participate in a very early stage.

Looking forward to see if there is any other better idea/suggestion for this

Reply:
And extend the load() function to support only loading this subset of configuration fields.

Reply:
If other fields reference vault, it will still have problem. We also need to extend load() to support only loading this subset of configuration fields.

Reply:
@windmgc have solved this problem by defining a list of fields that can't reference vault and only loading a subset of configs that are impossible to reference vault at the stage of `bin/kong`. Please help re-review this.